Fix #6794: Add landing page for negative numbers and refactor landing page.

DubeySandeep opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes #6794:  Add landing page for negative numbers and refactored landing page.
Work in this PR:
- Redesign orientation for the landing page.
- Adds `alt` for images.
- Adds topic in landing pages.
- Adds negative number landing pages.
 
## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

DubeySandeep committed:
Added landing page for negative numbers and refactored landing page.

DubeySandeep commented:
##  Old vs new UI changes:

![compare](https://user-images.githubusercontent.com/16653571/58906620-f4ae8e00-8729-11e9-913a-de6bd0a0abd9.png)
## Landing page for negative numbers:

![image](https://user-images.githubusercontent.com/16653571/58906689-1a3b9780-872a-11e9-922c-16c67ff7f243.png)


## Topics in landing page:

![image](https://user-images.githubusercontent.com/16653571/58906737-32131b80-872a-11e9-8eaf-8d6d8eddcb28.png)

DubeySandeep committed:
Add css property in sorted order.

DubeySandeep committed:
Undo changes in packe-lock.json

DubeySandeep commented:
@seanlip, should I upload the compressed version "Negative-number" video? 
Video size:
Fractions: 319KB
Ratios: 245KB
**Negative-numbers: 1.2MB**

seanlip commented:
Thanks @DubeySandeep! Some quick notes:

- "Negative-numbers just got easier" -- please remove the hyphen.
- Yes, let's use a compressed video.
- The changes to the landing page look great, thanks!

seanlip commented on core/templates/dev/head/pages/landing/TopicLandingPage.ts:
```diff
@@ -29,17 +31,61 @@ oppia.constant('TOPIC_LANDING_PAGE_DATA', {
+        },
         video: 'ratios_video.mp4',
+        lessons: [
+          'What is Ratio?',
```
What is a Ratio?

DubeySandeep replied:
Done!

seanlip commented on core/templates/dev/head/pages/landing/TopicLandingPage.ts:
```diff
@@ -29,17 +31,61 @@ oppia.constant('TOPIC_LANDING_PAGE_DATA', {
+        video: 'negative-numbers_video.mp4',
+        lessons: [
+          'The number line',
+          'What is a negative number?',
```
Use capitalized case: The Number Line, What is a Negative Number?, etc. -- for consistency.

DubeySandeep replied:
Done!

seanlip reviewed (changes requested):
Thanks @DubeySandeep, this is looking great!

For the new "topics" section, could you please follow the mocks @dchen97 posted in #6794 (feel free to ask her if you have any questions)? In particular use the correct colours to separate the various topics and the "and more". Also I would suggest one amendment, let's change "and more!" to "... and more!" to differentiate it from the ones above.

Thanks!

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into negative-landing

DubeySandeep committed:
Add hypenstospaces filter and fix typos.

DubeySandeep committed:
Change landing_page_data structure to include topic_title.

DubeySandeep committed:
Update with upstream/develop and resolved merge conflicts.

DubeySandeep commented:
@seanlip, I've addressed your comments PTAL! 
## Updated images:
![image](https://user-images.githubusercontent.com/16653571/59157462-abaf5e80-8ac8-11e9-9cb5-3d3cb806e5d1.png)
![image](https://user-images.githubusercontent.com/16653571/59157465-be299800-8ac8-11e9-83df-f243be25d8b0.png)
![image](https://user-images.githubusercontent.com/16653571/59157468-cf72a480-8ac8-11e9-8435-730d6e40f2d2.png)
![image](https://user-images.githubusercontent.com/16653571/59157472-e1ecde00-8ac8-11e9-90e6-cf8696ba78c2.png)
![image](https://user-images.githubusercontent.com/16653571/59157476-eca77300-8ac8-11e9-8e66-6aed829937f1.png)

seanlip commented:
Hm, tests are failing?

DubeySandeep committed:
Added proper filename.

apb7 commented:
Hi @DubeySandeep and @seanlip, I took a quick look at the logs and found this error:
```
ERROR in ./core/templates/dev/head/pages/landing/TopicLandingPage.ts
Module not found: Error: Can't resolve 'components/background/BackgroundBannerDirective.ts' in '/home/circleci/oppia/core/templates/dev/head/pages/landing'
 @ ./core/templates/dev/head/pages/landing/TopicLandingPage.ts 17:0-61
``` 
Let me know if this helps. Thanks!

DubeySandeep commented:
@seanlip, I've fixed the test issue (introduced after updating with upstream/develop).

DubeySandeep commented:
@apb7, Yeah, I've fixed the issue. Thanks for the help! :)

seanlip reviewed (approved):
LGTM. Thanks @DubeySandeep!

vojtechjelinek commented on core/templates/dev/head/pages/landing/TopicLandingPage.ts:
```diff
@@ -16,8 +16,11 @@
+require(
+  'components/common-layout-directives/common-elements/' +
+  'background-banner.directive.ts');
+
```
Please remove this empty line.

DubeySandeep replied:
I was following the guide provided in the wiki page.
Link: https://github.com/oppia/oppia/wiki/Coding-style-guide#adding-require-to-the-typescript-files-with-directive

> The requires for directives that are in HTML are included first and separated from the regular requires by empty line.

DubeySandeep replied:
@vojtechjelinek, we aren't following this rule anymore?  or have I missed something? :p

vojtechjelinek replied:
Oh, my bad I didn't realize this is an HTML related require.

vojtechjelinek reviewed (approved):
LGTM! Just one small nit!

DubeySandeep committed:
Updated with upstream/develop and resolved merge conflicts.

DubeySandeep commented:
@nithusha21, can you please review this PR as a codeowner?

nithusha21 reviewed (approved):
LGTM as a codeowner.
